File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();

        if(numRows <= 0)
        {
            return triangle;
        }

        temp.add(1);
        triangle.add(temp);
        if(numRows != 1)
        {
            temp = new ArrayList<>();
            temp.add(1);
            temp.add(1);
            triangle.add(temp);
        }
        
        for(int i=2 ; i<numRows; i++)
        {
            temp = new ArrayList<>();
            int length = triangle.get(i-1).size();
            temp.add(1);
            for(int j=1 ; j<length ; j++)
            {
                temp.add(triangle.get(i-1).get(j-1) + triangle.get(i-1).get(j));
            }
            temp.add(1);
            triangle.add(temp);
        }
    
        return triangle;
    }
}

File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 1:
            return [[1]]
        if numRows == 2:
            return [[1], [1,1]]

        triangle = list()
        temp = list()

        triangle.append([1])
        triangle.append([1,1])

        for i in range(2, numRows):
            temp.append(1)
            for j in range(len(triangle[i - 1]) - 1):
                temp.append(triangle[i - 1][j] + triangle[i - 1][j + 1])
            temp.append(1)
            triangle.append(temp.copy())
            temp.clear()

        return triangle